src/middlewared/middlewared/plugins/zfs.py

```diff
@@ -88,6 +88,14 @@ def query(self, filters, options):
                 pools = [i.__getstate__(**state_kwargs) for i in zfs.pools]
         return filter_list(pools, filters, options)
 
+    def query_imported_fast(self):
```

Comment:
I think we should rename this method as it will get pools which are available right now in the system which includes any which have been imported or created. So `query_available_pools_guid` or something along these lines seems better to me.

Reply:
No, this is querying looks that have been imported.

Reply:
So a created pool == imported pool ? I don't mind the current name but it seems off. Looking apart from the import pool perspective, if i was to just get pools with their guids, it seemed off to me to use `query_imported_fast` method name

Reply:
I'm not sure what "created" means in this context. Sure, the pools that have been listed here are also "created" but again, not sure what that means.

This list pools that have been imported only. 
```
with libzfs.ZFS() as zfs:
    print({i.guid: i.name for i in zfs.pools})
```

If you want to look at "created" pools (but not imported)"
```
with libzfs.ZFS() as zfs:
    for i in zfs.find_import():
        print(i)
```